theseus/geometry/functions/so3_function.py

```diff
@@ -0,0 +1,202 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+import theseus
+
+from typing import Optional, List, cast
+
+from .lie_group_function import LieGroupExpMap
+from .utils import check_jacobians_list
+
+
+def name() -> str:
+    return "SO3"
+
+
+def dim() -> int:
+    return 3
+
+
+def check_group_tensor(tensor: torch.Tensor) -> bool:
+    with torch.no_grad():
+        if tensor.ndim != 3 or tensor.shape[1:] != (3, 3):
+            raise ValueError("SO3 data tensors can only be 3x3 matrices.")
+
+        MATRIX_EPS = theseus.constants._SO3_MATRIX_EPS[tensor.dtype]
+        if tensor.dtype != torch.float64:
+            tensor = tensor.double()
+
+        _check = (
+            torch.matmul(tensor, tensor.transpose(1, 2))
+            - torch.eye(3, 3, dtype=tensor.dtype, device=tensor.device)
+        ).abs().max().item() < MATRIX_EPS
+        _check &= (torch.linalg.det(tensor) - 1).abs().max().item() < MATRIX_EPS
+
+    return _check
+
+
+def check_tangent_vector(tangent_vector: torch.Tensor) -> bool:
+    _check = tangent_vector.ndim == 3 and tangent_vector.shape[1:] == (3, 1)
+    _check |= tangent_vector.ndim == 2 and tangent_vector.shape[1] == 3
+    return _check
+
+
+class ExpMap(LieGroupExpMap):
+    @classmethod
+    def call(
+        cls,
+        tangent_vector: torch.Tensor,
+        jacobians: Optional[List[torch.Tensor]] = None,
+    ) -> torch.Tensor:
+        if not check_tangent_vector(tangent_vector):
+            raise ValueError("Tangent vectors of SO3 should be 3-D vectors.")
+        tangent_vector = tangent_vector.view(-1, 3)
+        ret = tangent_vector.new_zeros(tangent_vector.shape[0], 3, 3)
```

Comment:
I am assuming the math is same as out current implementation and therefore functorch compatible? We might want to include some unit tests related to vectorization as well for this api.

Reply:
Yes, it is functorch compatible.